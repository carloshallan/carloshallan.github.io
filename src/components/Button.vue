<template>
  <button :class="{ outlined }" @click="handleClick">
    <slot />
  </button>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'Button',
  props: {
    click: Function,
    outlined: Boolean
  },
  methods: {
    handleClick(event: HTMLElement) {
      this.click(event)
    }
  }
})
</script>

<style lang="stylus" scoped>
@import '../styles/default'

button
  background dark-purple
  border-radius 15px
  border 2px solid green
  text-transform uppercase
  color white
  padding 8px 36px
  font-size 20px
  FiraCode()
  transition 0.3s

  &.outlined
   color green
   background-color transparent

  &:hover
   background-color light-pink
   border-color pink
   cursor pointer

  &.outlined:hover
   color light-pink
   background-color transparent
</style>
