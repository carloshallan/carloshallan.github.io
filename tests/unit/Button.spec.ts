import { shallowMount } from '@vue/test-utils'
import Button from '@/components/Button.vue'

describe('Button.vue', () => {
  it('Render button with children', () => {
    const text = 'Click Here'
    const button = shallowMount(Button, {
      slots: { default: text }
    })
    expect(button.text()).toContain(text)
  })
})
