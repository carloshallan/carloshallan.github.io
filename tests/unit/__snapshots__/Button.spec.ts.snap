// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue Render button with children 1`] = `
<button
  class=""
>
  
  Click Here
  
</button>
`;
